Source/Scene/Cesium3DTileBatchTable.js

```diff
@@ -201,6 +201,9 @@ Cesium3DTileBatchTable.prototype.getShow = function (batchId) {
 };
 
 Cesium3DTileBatchTable.prototype.setColor = function (batchId, color) {
+  if (!defined(color)) {
+    return;
+  }
```

Comment:
One problem with this approach is that if a new style is applied any features that fall into the "else" condition will stay the same color that they were given previously. I think the safer approach is to make the change directly in the `ConditionsExpression` code. 

```js
ConditionsExpression.prototype.evaluateColor = function (feature, result) {
  var conditions = this._runtimeConditions;
  if (!defined(conditions)) {
    return undefined;
  }
  var length = conditions.length;
  for (var i = 0; i < length; ++i) {
    var statement = conditions[i];
    if (statement.condition.evaluate(feature)) {
      return statement.expression.evaluateColor(feature, result);
    }
  }

  return Color.clone(Color.WHITE, result); // <------ see here
};
```
[Sandcastle](http://localhost:8080/Apps/Sandcastle/index.html#c=lVTbbtswDP0VIhhgB3DlpJcV7ZJiaPq4ocUabA91HxSbiYXIUiDJLrIi/z7KNzS9bBlg2DLFc3jEQ7viBiqBT2hgCgqfYIZWlAX7WcfCZJDW7zOtHBcKTTKI4DlRAA6Nocid0ZXI0Fx2wNQgd/hLG5nNm5RwGCVqN/ySqERVVM4JiRYd1duD3NriuhQyE2plQ5/dyGI2RYVsY0QhnKjQMp5lYcvRkr5MTHmBhrOl3M51WCvN0DqhuBNa9Spn3DhacXXClkYXN7gyiDY8Oj9lo/FFBKcj9vlifBzB+dnIywfQRiD1oGGpeQFy5F5uz/qdu5w5/YOiXNnwuIUCbIRL84/Sjrq8XX3PStNWGbFR37j2wMy6rcR9q5rHyc2cUu79dnPsVEtNtpCDfhEGBrOAuAY95T1XWcqtk+hbOtdaLri5Lp3Tinyf5VytEOp63vRlqVIvKxy2/v+3IO/EkmbI9v0DKMh4I7j0Oj89L2kOSoMPwaIdhMtuP3jckYgGtWuf7QF7rlSrTHiJVOChCwItibnj2cF0SlpLKf2Z+t485cJhMEwGj9FfgcFKcmuDPaxdbxeyxCAiv84OoCCZNPBUbo+F5m97iICFEel6HyqUn6Pa3n/jLbn7qrQUq9xZSXmHiijQkSVvSQ7FW4f4Hr6O1818RRLHRONMiR+5Rhkwz4UFulyOgNIi0HBjx/G4Nzz1F1DfBtFgUo/wVZf5VRQbbRyURoaMxQ6Lje+NjRdluqaBT631QJ86iV9CJ5moQGTTd/6ZkPq5oZ0lTd69+E3nuJrElP8GKnX9S7mt0Ei+9Wn5+OpbE2SMTWJ6fR/pmg/4FfMf)

Before|![3](https://user-images.githubusercontent.com/915398/123181326-18e97e80-d45b-11eb-8a80-18a940e74d03.jpg)|![4](https://user-images.githubusercontent.com/915398/123181325-18e97e80-d45b-11eb-9cad-be08a74b138a.jpg)
--|--|--
After|![3](https://user-images.githubusercontent.com/915398/123181326-18e97e80-d45b-11eb-8a80-18a940e74d03.jpg)|![2](https://user-images.githubusercontent.com/915398/123181324-18e97e80-d45b-11eb-98ea-ac80e050911a.jpg)

Reply:
Also the tests that were removed can be re-added.

Reply:
I've been thinking about how to fix this for `show` too and I don't think it's as simple to fix in `ConditionsExpression.prototype.evaluate`. Maybe the fix needs to go in `Cesium3DTileBatchTable.prototype.applyStyle`

[`show` sandcastle](http://localhost:8080/Apps/Sandcastle/index.html#c=dVJda9swFP0rlzCIC66ctN1KsySMpY8bHWvYHuo8KPZNLCZL4Up26Er++6781XbrwMjSveccHR2plgS1wiMSLMDgEVboVFWKH00tSkdZs15Z46UySOkohqfUAHgk4so3srXKkWY9MSOUHn9a0vm6hURncWpOZx9Tk5qat/NKo0PP+72i3Lnyc6V0rszeRQHd2hIuQ4PiQKpUXtXohMzzqNPoRF8CM1kiSbHTj2sbNU5zdF4Z6ZU1g8uVJM8zaS7Fjmx5i3tCdNH59ZWYTG9iuJqIDzfTixiu30+CfQBLCjmDVqXRBShQBruD6lfpC+Htd65K46KLjgpwUD4r/gc773GnZswr6naZiMkQXHdg4fyjxtdX1f4ub9cMuQ/t/tg7vjA3mC05Y1JSzyAdvXvaceQV4cN422U+6/vjzYkv+dmPK+xxEMmsyVVwx7oPbQl4woI9/QSLxQLGey2dG4fX4qnCTdxjk4TRofTc4hqsC+WAP18goHYImXTYcjaDlyaKUTyaNyEse8lPqjxY8lCRjoRIPJYHzWZcsq2yXxxZ5lwgBug8eUmd56oGlS/eeOSQBf/c2VVa36vfbHc5Txj/D1Xb5g3c1UhaPgZYMV1+aYtCiHnCy7eZ3lq9lfSX8h8)